File: Queues/CircularQueue.cs

using System.Reflection;
class MainProgram
{
    public static void Main(string[] args)
    {
        CircularQueue<string> circ = new CircularQueue<string>(5);

        circ.Enqueue("1");
        circ.Enqueue("2");
        circ.Enqueue("3");
        circ.Enqueue("4");
        circ.Enqueue("5");
        circ.Enqueue("6");
        circ.DeQueue();
        circ.DeQueue();
        circ.Enqueue("6");
        circ.Enqueue("9");
        circ.Enqueue("69");
        circ.Enqueue("9887");
        circ.Enqueue("643");
        circ.Enqueue("953553");
        circ.DeQueue();
        circ.IsEmpty();
        circ.Enqueue("3jk");
        circ.IsFull();
        circ.DeQueue();
        circ.DeQueue();
        circ.DeQueue();
        circ.DeQueue();
        circ.DeQueue();
        circ.IsFull();
        circ.IsEmpty();
    }
}

// Circular Queue
public class CircularQueue<J>
{
    public int Maxs;
    public int FP;
    public int RP;
    public int count;

    private J[] Data;

    public CircularQueue(int Max)
    {
        Data = new J[Max];
        Maxs = Max
    ;
        RP = -1;
        count = 0;
    }

    public void Enqueue(J i)
    {
        if (count == Maxs)
        {
        Console.WriteLine("error Queue is full");
        }   
        else if (RP == Maxs-1 && FP > 0)
        {
            Console.WriteLine("[Loop]Adding " + i);
            RP = 0;
            Data[RP] = i;
            count += 1;
        }
        else
        {
            Console.WriteLine("Adding " + i);
            RP += 1;
            Data[RP] = i;
            count += 1;
        }
    }
    public void peek()
    {
        Console.WriteLine("The Current Task Is " + Data[FP]);
    }
    public void DeQueue()
    {
        if (FP == Maxs-1 && RP > 0)
        {
            Console.WriteLine("[Loop]Executing " + Data[FP]);
            FP = 0;
            count -= 1;
        }
        else
        {
            Console.WriteLine("executing " + Data[FP]);
            FP += 1;
            count -= 1;
        }
    }
    public void IsFull()
    {
        if (count == Maxs)
        {
            Console.WriteLine("True");
        }
        else
        {
            Console.WriteLine("False");
        }
    }
    public void IsEmpty()
    {
        if (count == 0)
        {
            Console.WriteLine("True");
        }
        else
        {
            Console.WriteLine("False");
        }
    }
}
    

File: Queues/Python/CircularQueue.py
class CircularQueue():
    def __init__(self, Maxs):
        self.Maxs = Maxs
        self.Count = 0
        self.Data = []
        self.RP = -1
        self.FP = 0
        
    
    def EnQueue(self,Item):
        if self.Count == self.Maxs:
            print("error queue is full")
        elif self.RP == self.Maxs-1 and self.FP > 0:
            print("[loop]Adding ", Item)
            self.RP = 0
            self.Data[self.RP] = Item
            self.Count += 1
        else:
            print("Adding ", Item)
            self.RP += 1
            self.Data.append(Item)
            self.Count += 1

    def Peek(self):
        print("The Current Task Is ", self.Data[self.FP])
    
    def DeQueue(self):
        if self.FP == self.Maxs-1 and self.RP > 0:
            print("[Loop]Executing ", self.Data[self.FP])
            self.FP = 0
            self.Count -= 1
        else:
            print("executing ", self.Data[self.FP])
            self.FP += 1
            self.Count -=1

    def IsFull(self):
        if self.Count == self.Maxs:
            print("True")
        else:
            print("False")

    def IsEmpty(self):
        if self.Count == 0:
            print("True")
        else:
            print("False")

Circ = CircularQueue(5)

Circ.EnQueue("1");
Circ.EnQueue("2");
Circ.EnQueue("3");
Circ.EnQueue("4");
Circ.EnQueue("5");
Circ.EnQueue("6");
Circ.DeQueue();
Circ.DeQueue();
Circ.EnQueue("6");
Circ.EnQueue("9");
Circ.EnQueue("69");
Circ.EnQueue("9887");
Circ.EnQueue("643");
Circ.EnQueue("953553");
Circ.DeQueue();
Circ.IsEmpty();
Circ.EnQueue("3jk");
Circ.IsFull();
Circ.DeQueue();
Circ.DeQueue();
Circ.DeQueue();
Circ.DeQueue();
Circ.DeQueue();
Circ.IsFull();
Circ.IsEmpty();